File: java/src/main/java/cloud/cniia/openapi/sdk/ApiException.java
package cloud.cniia.openapi.sdk;

import java.util.Map;

/**
 * 业务异常：统一响应信封中 {@code code != 0} 时抛出。
 *
 * <p>携带原始 {@code code}/{@code message}/{@code data} 与原始响应体文本，供调用方判断。
 * 不要穷举 code 写死分支——错误码以服务端为准，可能新增（见 INTERFACES.md §六）。
 */
public class ApiException extends RuntimeException {

    private final long code;
    private final Object data;
    private final String rawBody;

    public ApiException(long code, String message, Object data, String rawBody) {
        super("OpenAPI 业务错误 code=" + code + " message=" + message);
        this.code = code;
        this.data = data;
        this.rawBody = rawBody;
    }

    /** 业务错误码（非 0）。 */
    public long code() {
        return code;
    }

    /** data 字段（可能为 null，或含 missing_fields 等）。 */
    public Object data() {
        return data;
    }

    /** data 转 Map（不是 Map 时返回 null）。 */
    @SuppressWarnings("unchecked")
    public Map<String, Object> dataAsMap() {
        return data instanceof Map ? (Map<String, Object>) data : null;
    }

    /** 原始响应体文本（信封完整 JSON）。 */
    public String rawBody() {
        return rawBody;
    }
}

File: php/src/Exception/ApiException.php
<?php

declare(strict_types=1);

namespace ProjectP\Sdk\Exception;

/**
 * 业务异常：统一信封 {code,message,data} 中 code !== 0 时抛出。
 *
 * 携带原始 code / message / data，供调用方按错误码判断（不要穷举写死分支）。
 */
final class ApiException extends \RuntimeException
{
    /**
     * @param int                       $apiCode 业务错误码（非 0）
     * @param string                    $apiMessage 业务消息
     * @param array<string,mixed>|null  $data    业务 data（可能含 missing_fields 等）
     * @param array<string,mixed>|null  $raw     完整原始响应信封
     */
    public function __construct(
        public readonly int $apiCode,
        public readonly string $apiMessage,
        public readonly ?array $data = null,
        public readonly ?array $raw = null,
    ) {
        parent::__construct(
            sprintf('OpenAPI 业务错误 [%d]: %s', $apiCode, $apiMessage),
            $apiCode
        );
    }
}
